### PR TITLE
chore: dimension select refactor

### PR DESCRIPTION
**Pull request by turban**
This PR will refactor `DimensionSelect` and `DimensionItemsSelect` components to use:
- Functional component
- DHIS2 app runtime (useDataQuery)
- Load data in component (skip redux)

Screenshot 2023-01-16 at 11 41 44

### CODE REVIEW

File: src/components/dimensions/DimensionItemsSelect.js [hunk 1]
```diff
@@ -1,57 +1,65 @@
...
-        if (dimension && !items) {
-            loadDimensionItems(dimension)
+// Load dimension items
+const DIMENSION_ITEMS_QUERY = {
+    items: {
+        resource: 'dimensions',
+        id: ({ id }) => `${id}/items`,
+        params: ({ nameProperty }) => ({
+            fields: ['id', `${nameProperty}~rename(name)`],
+            order: 'displayName:asc',
```
Comment (jenniferarnesen) [inline src/components/dimensions/DimensionItemsSelect.js:16]: Will `order` work when it's always `displayName`?

File: src/components/dimensions/DimensionSelect.js [hunk 2]
```diff
@@ -1,32 +1,42 @@
...
-        }
-    }, [dimensions, loadDimensions])
+// Load above dimension types
+const DIMENSIONS_QUERY = {
+    dimensions: {
+        resource: 'dimensions',
+        params: ({ nameProperty }) => ({
+            fields: ['id', `${nameProperty}~rename(name)`, 'dimensionType'],
+            filter: `dimensionType:in:[${dimensionTypes.join(',')}]`,
+            order: 'displayName:asc',
```
Comment (jenniferarnesen) [inline src/components/dimensions/DimensionSelect.js:24]: also here, will `displayName` always work?

File: src/components/dimensions/DimensionItemsSelect.js (hunk 1 above)
Comment (turban) [reply src/components/dimensions/DimensionItemsSelect.js:16]: Changed to `${nameProperty}:asc`

File: src/components/dimensions/DimensionSelect.js (hunk 2 above)
Comment (turban) [reply src/components/dimensions/DimensionSelect.js:24]: Changed to `${nameProperty}:asc`

**Pull request merged**